src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -190,7 +190,20 @@ private static SafeFileHandle CreateSharedBackingObject(Interop.Sys.MemoryMapped
             do
             {
                 mapName = GenerateMapName();
-                fd = Interop.Sys.ShmOpen(mapName, flags, (int)perms); // Create the shared memory object.
+                if (Interop.Sys.MemfdSupported)
+                {
+                    fd = Interop.Sys.MemfdCreate(mapName);
+                    // Use mmap to set protection
+                    IntPtr mappedRegion = Interop.Sys.MMap(IntPtr.Zero, (ulong)capacity, protections, Interop.Sys.MemoryMappedFlags.MAP_SHARED, fd, 0);
```

Comment:
mmap is already being called in CreateView. Why do we need to call it here? And why would we then ignore the returned pointer rather than later using it with munmap?

Reply:
Ah, it's already called in `MemoryMappedView.Unix.cs`. Will revert this change.